workitem/link/link_repository.go

```diff
@@ -108,9 +120,35 @@ func (r *GormWorkItemLinkRepository) Create(ctx context.Context, sourceID, targe
 	if err := link.CheckValidForCreation(); err != nil {
 		return nil, errs.WithStack(err)
 	}
-	if err := r.ValidateCorrectSourceAndTargetType(ctx, sourceID, targetID, linkTypeID); err != nil {
+
+	// Fetch the link type
+	linkType, err := r.workItemLinkTypeRepo.LoadTypeFromDBByID(ctx, linkTypeID)
+	if err != nil {
+		return nil, errs.Wrap(err, "failed to load link type")
+	}
+
+	if err := r.ValidateCorrectSourceAndTargetType(ctx, sourceID, targetID, linkType); err != nil {
 		return nil, errs.WithStack(err)
 	}
+
+	// check to disallow multiple parents in tree topology
+	if linkType.Topology == TopologyTree {
+		parentExists, err := r.CheckParentExists(ctx, targetID, linkType)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"link_type_id": linkTypeID,
```

Comment:
Please, use `wilt_id` instead of `link_type_id` as done in our parts of our code.